src/CaptureFile/CaptureSectionInputStreamImpl.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureSectionInputStreamImpl.h"
+
+#include "OrbitBase/MakeUniqueForOverwrite.h"
+
+namespace orbit_capture_file_internal {
+using orbit_grpc_protos::ClientCaptureEvent;
+
+ErrorMessageOr<ClientCaptureEvent> CaptureSectionInputStreamImpl::ReadEvent() {
+  uint64_t message_size = 0;
+
+  if (!coded_input_stream_.ReadVarint64(&message_size)) {
+    return file_fragment_input_stream_.GetLastError().value_or(
```

Comment:
It feels to me that I a missing a small bit. Why is `file_fragment_input_stream_` only used for reading the last error?
How does it correlate with the `coded_input_stream_`?
Could you add a comment somewhere to describe the relationship (could be in the class level comment)?

Reply:
Edit: Okay after reading the header again, I see that coded_input_stream_ is backed by the file_fragment_input_stream_. Still think that a brief comment could help.

Reply:
done.